groups/script.js

```diff
@@ -178,25 +202,42 @@ searchInput.addEventListener('keyup', () => {
  * TO ASSIGN YOURSELF A ROLE
  */
 const buttonAddRole = document.querySelector('.btn-add-role');
-buttonAddRole.addEventListener('click', async function () {
+async function addrole() {
   if (memberAddRoleBody?.userid && memberAddRoleBody?.roleid !== '') {
     loader.classList.remove('hidden');
 
-    addGroupRoleToMember(memberAddRoleBody)
-      .then((res) => {
-        const groupNameElement = document.getElementById(
-          `name-${memberAddRoleBody.roleid}`,
-        );
-        const currentCount = groupNameElement.getAttribute('data-member-count');
-        if (currentCount !== null && currentCount !== undefined) {
-          groupNameElement.setAttribute('data-member-count', +currentCount + 1);
-        }
-        alert(res.message);
-      })
-      .catch((err) => alert(err.message))
-      .finally(() => loader.classList.add('hidden'));
+    try {
+      // Add the role to the member
+      const res = await addGroupRoleToMember(memberAddRoleBody);
+
+      const groupNameElement = document.getElementById(
+        `name-${memberAddRoleBody.roleid}`,
+      );
+      const currentCount = groupNameElement.getAttribute('data-member-count');
+      if (currentCount !== null && currentCount !== undefined) {
+        groupNameElement.setAttribute('data-member-count', +currentCount + 1);
+      }
+      alert(res.message);
+
+      // After adding the role, re-fetch the user group data to update it
+      UserGroupData = await getUserGroupRoles();
+
+      // Update the button state with the refreshed data
+      updateButtonState();
+    } catch (err) {
+      alert(err.message);
+    } finally {
+      loader.classList.add('hidden');
+    }
   }
-});
+}
+
+/**
+ * TO REMOVE YOURSELF OF A ROLE
+ */
+async function removeRoleHandler() {
+  console.log('Remove function to be added after this pr');
```

Comment:
why console log here

Reply:
Actually the remove feature is not implemented yet only the UI part is done. That's why for the time being written console log temporarily for the remove button when clicked.
Feature to be implemented after this pr as planned as seperated my task into different issues
Isssu ticket : 
- #482